server/src/main/java/com/openelements/benchscape/server/store/endpoints/BenchmarkEnvironmentEndpoint.java

```diff
@@ -97,24 +99,26 @@ public Environment find(@RequestParam final String id) {
      * @return all environments that match the given query parameters
      */
     @GetMapping("/findByQuery")
-    public List<Environment> findByQuery(@RequestParam(required = false) String name,
+    public List<Environment> findByQuery(
+            @RequestParam(required = false) String name,
             @RequestParam(required = false) String gitOriginUrl,
             @RequestParam(required = false) String gitBranch,
             @RequestParam(required = false) String systemArch,
             @RequestParam(required = false) Integer systemProcessors,
             @RequestParam(required = false) Integer systemProcessorsMin,
             @RequestParam(required = false) Integer systemProcessorsMax,
-            @RequestParam(required = false) Long systemMemory,
-            @RequestParam(required = false) Long systemMemoryMin,
-            @RequestParam(required = false) Long systemMemoryMax,
+            @RequestParam(required = false) SystemMemory systemMemory,
```

Comment:
I assume that will not work quite well. In the endpoint methods that can be called by REST I would suggest to always use `Long systemMemory`.

Reply:
Can be done in future PR